ios/RNShopify.m

```diff
@@ -0,0 +1,320 @@
+        }
+
+        resolve(collectionDictionaries);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductTags:(NSUInteger)page resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductTagsPage:page completion:^(NSArray<NSString *> *tags, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve(tags);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductsPage:(NSUInteger)page resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductsPage:page completion:^(NSArray<BUYProduct *> *products, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductsWithTags:(NSUInteger)page tags:(NSArray<NSString *> *)tags resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductsByTags:tags page:page completion:^(NSArray<BUYProduct *> *products, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductsWithTagsForCollection:(NSUInteger)page collectionId:(nonnull NSNumber *)collectionId tags:(NSArray<NSString *> *)tags resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductsPage:page inCollection:collectionId withTags:tags sortOrder:BUYCollectionSortCollectionDefault completion:^(NSArray<BUYProduct *> *products, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(webCheckout:(NSArray *)cart resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
+{
+    _resolve = resolve;
+    _reject = reject;
+
+    BUYCheckout *checkout = [self createCheckoutFromCart:cart];
+
+    [self.client createCheckout:checkout completion:^(BUYCheckout *checkout, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        BUYWebCheckoutPaymentProvider *webPaymentProvider = [[BUYWebCheckoutPaymentProvider alloc] initWithClient:self.client];
+        webPaymentProvider.delegate = self;
```

Comment:
Ok, but we should be careful with this. What happens for example if a user starts another checkout after the first one is finished? Will we reuse the old checkout, and maybe introduce some weird bugs because of that?